File: src/main/java/com/piveguyz/empickbackend/employment/applicant/command/application/dto/ApplicationCommandDTO.java
package com.piveguyz.empickbackend.employment.applicant.command.application.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ApplicationCommandDTO {
    private Integer id;
    private int recruitmentId;
    private int applicantId;
    private int status;
    private Integer introduceRatingResultId;
}

File: Empick-FE/src/dto/employment/applicant/applicationCommandDTO.js
export default class ApplicationCommandDTO {
  constructor(id, status, introduceRatingResultId) {
    this.id = id;
    this.status = status;
    this.introduceRatingResultId = introduceRatingResultId;
  }

  static fromJSON(json) {
    return new ApplicationCommandDTO(
      json.id,
      json.status,
      json.introduceRatingResultId || json.introduce_rating_result_id
    );
  }

  static toJSON(dto) {
    return {
      id: dto.id,
      status: dto.status,
      introduceRatingResultId: dto.introduceRatingResultId,
      introduce_rating_result_id: dto.introduceRatingResultId // 백엔드 호환성
    };
  }
}
